ibm/resource_ibm_network_gateway.go

```diff
@@ -0,0 +1,816 @@
+	}
+
+	portSpeed, err := findNetworkItemPriceId(items, d)
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+
+	monitoring, err := getItemPriceId(items, "monitoring", "MONITORING_HOST_PING")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+	if d.Get("tcp_monitoring").(bool) {
+		monitoring, err = getItemPriceId(items, "monitoring", "MONITORING_HOST_PING_AND_TCP_SERVICE")
+		if err != nil {
+			return datatypes.Container_Product_Order{}, err
+		}
+	}
+	// Other common default options
+	priIpAddress, err := getItemPriceId(items, "pri_ip_addresses", "1_IP_ADDRESS")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+
+	pri_ipv6_addresses, err := getItemPriceId(items, "pri_ipv6_addresses", "1_IPV6_ADDRESS")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+
+	remoteManagement, err := getItemPriceId(items, "remote_management", "REBOOT_KVM_OVER_IP")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+	vpnManagement, err := getItemPriceId(items, "vpn_management", "UNLIMITED_SSL_VPN_USERS_1_PPTP_VPN_USER_PER_ACCOUNT")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+
+	notification, err := getItemPriceId(items, "notification", "NOTIFICATION_EMAIL_AND_TICKET")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+	response, err := getItemPriceId(items, "response", "AUTOMATED_NOTIFICATION")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+	vulnerabilityScanner, err := getItemPriceId(items, "vulnerability_scanner", "NESSUS_VULNERABILITY_ASSESSMENT_REPORTING")
+	if err != nil {
+		return datatypes.Container_Product_Order{}, err
+	}
+
+	// Define an order object using basic paramters.
+
+	order := datatypes.Container_Product_Order{
+		ContainerIdentifier: sl.String(d.Get("hostname").(string)),
+		Quantity:            sl.Int(1),
+		Hardware: []datatypes.Hardware{{
+			Hostname: sl.String(d.Get("hostname").(string)),
+			Domain:   sl.String(d.Get("domain").(string)),
+		},
+		},
+		Location:  sl.String(strconv.Itoa(*dc.Id)),
+		PackageId: pkg.Id,
+		Prices: []datatypes.Product_Item_Price{
+			server,
+			os,
+			ram,
+			portSpeed,
+			priIpAddress,
+			pri_ipv6_addresses,
+			remoteManagement,
+			vpnManagement,
+			monitoring,
+			notification,
+			response,
+			vulnerabilityScanner,
+		},
+	}
+
+	// Add optional price ids.
+	// Add public bandwidth
+
+	if publicBandwidth, ok := d.GetOk("public_bandwidth"); ok {
+		publicBandwidthStr := "BANDWIDTH_" + strconv.Itoa(publicBandwidth.(int)) + "_GB"
+		bandwidth, err := getItemPriceId(items, "bandwidth", publicBandwidthStr)
+		if err != nil {
+			return datatypes.Container_Product_Order{}, err
+		}
+		order.Prices = append(order.Prices, bandwidth)
+	}
+
+	// Add prices of disks.
+	var arrayDrives []interface{}
+	var arrayDrivestemp []string
+	arrayDrivestemp = append(arrayDrivestemp, "HARD_DRIVE_1_00_TB_SATA_III")
```

Comment:
Should this be exposed in schema as `disk_names` or something of that sort like in `ibm_bare_metal`

Reply:
Added this note into the documentation: RAID_CONTROLLER - (reference) this version support a single RAID 1 controller pre configured with two hard drives 1 TB SATA drives, the idea is to keep the Network Gateway with minimal internal storage configuration since it’s primary purpose is to serve the network